pandas/core/computation/parsing.py

```diff
@@ -0,0 +1,154 @@
+    # toke.tok_name contains a readable description of the replacement string.
+    special_characters_replacements = {
+        char: "_{}_".format(token.tok_name[tokval])
+        # The ignore here is because of a bug in mypy that is resolved in 0.740
+        for char, tokval in tokenize.EXACT_TOKEN_TYPES.items()  # type: ignore
+    }
+    special_characters_replacements.update(
+        {
+            " ": "_",
+            "?": "_QUESTIONMARK_",
+            "!": "_EXCLAMATIONMARK_",
+            "$": "_DOLLARSIGN_",
+            "€": "_EUROSIGN_",
+            # Including quotes works, but they have to come in pairs.
+            "'": "_SINGLEQUOTE_",
+            '"': "_DOUBLEQUOTE_",
+            # Currently not possible. Terminates parser and won't find backtick.
+            # "#": "_HASH_",
+        }
+    )
+
+    name = "".join(special_characters_replacements.get(char, char) for char in name)
+    name = "BACKTICK_QUOTED_STRING_" + name
+
+    if not name.isidentifier():
+        raise ValueError(
+            "Could not convert {} to a valid Python identifier.".format(name)
+        )
+
+    return name
+
+
+def tokenize_and_clean_column_names(name: Union[str, int]) -> Union[str, int]:
+    """Function to emulate the cleaning of a backtick quoted name.
+
+    The purpose for this function is to see what happens to the name of
+    identifier if it goes to the process of being parsed a Python code
+    inside a backtick quoted string and than being cleaned
+    (removed of any special characters).
+
+    Returns
+    -------
+    Returns the name after tokenizing and cleaning,
+    or 'name' if that operation failed, or is a number.
+    """
+    if not isinstance(name, str):
+        return name
+
+    tokenized = tokenize_string("`{}`".format(name))
+    tokval = next(tokenized)[1]
+
+    try:
+        return create_valid_python_identifier(tokval)
+    except ValueError:
+        # Rarely, a name cannot be converted to a valid Python identifier.
+        # So, we just return the name.
+        return name
+
+
+def tokenize_backtick_quoted_string(
+    token_generator: Iterator[tokenize.TokenInfo], source: str, string_start: int
+) -> Tok:
+    """Creates a token from a backtick quoted string.
+
+    Moves the token_generator forwards till right after the next backtick.
+    """
+    for _, tokval, start, _, _ in token_generator:
+        if tokval == "`":
+            string_end = start[1]
+            break
+
+    return BACKTICK_QUOTED_STRING, source[string_start:string_end]
+
+
+def tokenize_string(source: str) -> Iterator[Tok]:
+    """Tokenize a Python source code string.
+
+    Parameters
+    ----------
+    source : str
+        A Python source code string
+    """
+    line_reader = StringIO(source).readline
+    token_generator = tokenize.generate_tokens(line_reader)
+
+    # Loop over all tokens till a backtick (`) is found.
+    # Then, take all tokens till the next backtick to form a backtick quoted
```

Comment:
what happens if you only have 1 backtick? SyntaxError? is this tested?

Reply:
You will get an unreferenced variable error. I can catch this and re-throw a more readable ValueError. (SyntaxError is only for actual Python code.)

Reply:
Changed to SyntaxError as that is used elsewhere in the pandas code as well.